File: Java/442. Find All Duplicates in an Array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        HashMap<Integer, Integer> temp = new HashMap<>();
        List<Integer> ans = new ArrayList<>();

        for(int i: nums){
            if(temp.get(i) != null) ans.add(i);
            else temp.put(i, 1);
        }

        return ans;
    }
}

File: C++/442. Find All Duplicates in an Array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        map<int, int> m;
        vector<int> res;

        for(int i: nums) m[i]++;

        for(auto &[k, v]: m){
            if(v == 2) res.push_back(k);
        }

        return res;
    }
};
